### ISSUE TITLE
Request for additional documentation on learning rate scheduling on `step` instead of `epoch`.

### ISSUE THREAD

**Issue by veritas9872** [PyTorchLightning/pytorch-lightning#5675]
## 📚 Documentation

The current documentation states that returning `{'interval': 'step'}` in `configure_optimizers` will alter the learning rate scheduler update interval to step-wise update instead of epoch-wise update.

However, as mentioned in https://github.com/PyTorchLightning/pytorch-lightning/issues/4929, this is not true. I would like to request an update to the documentation, including a warning that the previous recommendation does not actually work.

**Comment by SkafteNicki** [comment]
Hi @veritas9872
Could you please check out the PR that I created (#5678) and see if clears up the confusion.

**Comment by veritas9872** [comment]
@SkafteNicki Hello. I am not sure if this is quite what I requested. The big problem in https://github.com/PyTorchLightning/pytorch-lightning/issues/4929 was that giving `{'interval': 'step'}` actually did *not* change the interval to a step-based interval but remained an epoch-based interval. Although fixing this bug would be the best option, the issue that I raised was closed with a suggestion for new options. 
The documentation update that I have seen in https://github.com/PyTorchLightning/pytorch-lightning/pull/5678/commits/1dc28b7d2f95c0b46f5bc9f3aa6c3f5bbd4ad31a do not mention this bug.
